### PR TITLE
Add documentation for AvAnon UAVCAN laser interface and air data computer

### PR DESCRIPTION
**Pull request by avionicsanonymous**
Adds documentation related to https://github.com/PX4/Firmware/pull/15097

- [ ] Waiting on https://github.com/PX4/Firmware/pull/15097

### CODE REVIEW

File: en/advanced_config/parameter_reference.md [hunk 1]
```diff
@@ -14037,6 +14037,22 @@ How often the sensor is readout</p>   <p><b>Reboot required:</b> true</p>
  <td style="vertical-align: top;">1</td>
  <td style="vertical-align: top;"></td>
 </tr>
+<tr>
```
Comment (hamishwillee) [inline en/advanced_config/parameter_reference.md:14040]: Please remove these changes. The parameters are updated automatically from firmware source code.

File: en/sensor/airspeed.md [hunk 2]
```diff
@@ -19,7 +19,9 @@ Recommended digital airspeed sensors include:
 * [Sensirion SDP3X Differential Pressure Sensor](https://www.sensirion.com/en/flow-sensors/differential-pressure-sensors/worlds-smallest-differential-pressure-sensor/)
 * [Holybro Digital Air Speed Sensor](https://shop.holybro.com/digital-air-speed-sensor_p1029.html)
 
-All the sensors are connected via the I2C bus/port.
+All the above sensors are connected via the I2C bus/port.
+
+> **Note** Additionally, the [Avionics Anonymous Air Data Computer](https://www.tindie.com/products/avionicsanonymous/uavcan-air-data-computer/) can be connected to the UAVCAN bus to determine the true static pressure (via air temperature measurement and a barometer).
```
Comment (hamishwillee) [inline en/sensor/airspeed.md:24]: I modified this a bit. Please confirm:
1. True static pressure information can be used with the other airspeed sensors to get better airspeed estimate. Is that the point of this?
2. Does PX4 do everything else needed to use this computer when it is attached? 
3. Does this also need a separate temperature measurement sensor and barometer, and are they provided?

Essentially confirm what is needed to make this work and why you bother.

File: en/sensor/avanon_laser_interface.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+## Where to Buy
+
+* [AvAnon Laser Interface](https://www.tindie.com/products/avionicsanonymous/uavcan-laser-altimeter-interface/)
+
+## Supported Rangefinders {#supported_rangefinders}
+
+A full list supported rangefinders can be found on the link above. 
+
+At time of writing the following rangefinders are supported:
+
```
Comment (hamishwillee) [inline en/sensor/avanon_laser_interface.md:14]: FYI, I added this list to make it more obvious what is supported. You might want to add a link to this page in http://docs.px4.io/master/en/sensor/sfxx_lidar.html

File: en/sensor/avanon_laser_interface.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+1   | POWER_OUT | Filtered power at the supply voltage.
+2   | CAN+ | TX for serial mode, Clock for I2C mode
+3   | RX/SDA | RX for serial mode, Data for I2C mode
+4   | GND | Signal/power ground.
+
+
+## Wiring
+
+The rangefinder (laser) is connected to the AvAnon interface board, which is connected to one of the CAN ports on your autopilot.
+The wiring is as per the pinout above or using the included cables.
```
Comment (hamishwillee) [inline en/sensor/avanon_laser_interface.md:47]: What cables are included? Worth saying you can connect these rangefinders/boards "out of the box"

File: en/sensor/rangefinders.md [hunk 5]
```diff
@@ -59,6 +59,9 @@ The [Benewake TFmini Lidar](../sensor/tfmini.md) is a tiny, low cost, and low po
 The [Lanbao PSK-CM8JL65-CC5 ToF Infrared Distance Measuring Sensor](../sensor/cm8jl65_ir_distance_sensor.md) is a very small (38 mm x 18mm x 7mm, <10g) IR distance sensor with a 0.17m-8m range and millimeter resolution.
 It must be connected to a UART/serial bus. 
 
+### Avionics Anonymous UAVCAN Laser Altimeter Interface 
+
+The [Avionics Anonymous UAVCAN Laser Altimeter Interface](../sensor/avanon_laser_interface.md) allows several common rangefinders to be connected to the UAVCAN bus, a more robust interface than I2C.
```
Comment (hamishwillee) [inline en/sensor/rangefinders.md:64]: Perhaps linking to the common rangefinders list and mentioning a few here - e.g. 

> ... allows several [common rangefinders] (e.g. Lightware SF xxx, etc.) to be connected to the UAVCAN bus, a more robust interface than I2C.

**Comment by hamishwillee** [comment]
Thanks very much. You will need to add sensor/avanon_laser_interface.md  to /en/SUMMARY.md for this to render.

File: en/sensor/airspeed.md (hunk 2 above)
Comment (avionicsanonymous) [reply en/sensor/airspeed.md:24]: Thanks for the improvements!

1 and 3. Not quite - this devices includes its own differential pressure sensor, barometer, and outside air temp probe. The static pressure measurement it provides can't be used to improve airspeed measured by a separate device, but the plumbed static source to the barometer onboard this device gives you a much more accurate static pressure source for high-accuracy altitude, as long as you plumb it to a good static pressure source on the vehicle, i.e. on an air data boom. Does that make sense?
2. Yes, it's plug and play as long as UAVCAN is enabled. PX4 will recognize it as both an airspeed or differential pressure source (can be configured as either) as well as a barometer.

File: en/sensor/avanon_laser_interface.md (hunk 4 above)
Comment (avionicsanonymous) [reply en/sensor/avanon_laser_interface.md:47]: Yes, good point. The unit can be purchased with the cables to connect to your laser and to the autopilot. I'll note that!

File: en/sensor/rangefinders.md (hunk 5 above)
Comment (avionicsanonymous) [reply en/sensor/rangefinders.md:64]: Will do!

**Comment by avionicsanonymous** [comment]
Updated to, I believe, address all your notes.

Should I add a dedicated page for the Air Data Computer and its setup, similar to the Laser Interface? It's straightforward, but does require enabling UAVCAN. I can include wiring and air data plumbing suggestions as well.

File: en/sensor/airspeed.md (hunk 2 above)
Comment (hamishwillee) [reply en/sensor/airspeed.md]: Great, the new words look good.

**Comment by hamishwillee** [comment]
@avionicsanonymous Thanks for fixes. The only thing still needed for this is that any new document must be listed in `en/SUMMARY.md`. This makes it appear in the sidebar - otherwise it will not be rendered.

> Should I add a dedicated page for the Air Data Computer and its setup, similar to the Laser Interface? It's straightforward, but does require enabling UAVCAN. I can include wiring and air data plumbing suggestions as well.

I think that would be really cool, but that is up to you. This PR itself could go in without that. 

This can go in once the associated firmware PR is ready.

**Comment by avionicsanonymous** [comment]
Added the AvAnon GNSS/Mag and Magnetometer to the GPS/Compass page.

**Comment by hamishwillee** [comment]
Thanks. I'm going to merge this now and do fixes as a post process (ie. file is not in summary.md so will not render.

**Pull request merged**